include/miopen/miopen.h

```diff
@@ -4667,6 +4667,119 @@ miopenReduceTensor(miopenHandle_t handle,
 /** @} */
 // CLOSEOUT TensorReduce DOXYGEN GROUP
 
+/**
+ * An miopenProblem will describe a problem for different miopen operations.
+ * For now, this is only used for convolution, but could be used for other
+ * operators in the future(such as GEMM, Pooling, etc)
+ */
+MIOPEN_DECLARE_OBJECT(miopenProblem);
+
+typedef enum
+{
+    miopenProblemDirectionForward        = 0,
+    miopenProblemDirectionBackward       = 1,
+    miopenProblemDirectionBackwardWeight = 2,
+} miopenProblemDirection_t;
+
+// Enums with no meaningful default value should probably have invalid default for fool-proofing.
+typedef enum
+{
+    miopenTensorNameInvalid  = 0,
+    miopenTensorConvolutionX = 1,
+    miopenTensorConvolutionW = 2,
+    miopenTensorConvolutionY = 3,
```

Comment:
Perhaps X, W, and Y are clear for most AI folks, although I think names like `miopenTensorConvolutionInput`, `miopenTensorConvolutionWeghts`, and `miopenTensorConvolutionOutput` would be clearer. What does everyone else think about the names?

Reply:
input/output terminology is unclear for different directions. For example, which is which for backward weights? Technically weights buffer would be the output of the operation.

Reply:
Ah yea makes sense.